troposphere/static/js/components/projects/resources/instance/details/sections/InstancePlaybookHistorySection.jsx

```diff
@@ -0,0 +1,140 @@
+        if(instance_playbook_histories == null) {
+            return (<tr>
+                        <td><span className="loading-tiny-inline"/></td>
+                    </tr>);
+        } else if (instance_playbook_histories.length == 0) {
+            return (
+                <tr>
+                    {"No commands found."}
+                </tr>
+            );
+        } else {
+            return instance_playbook_histories.map(this.renderInstancePlaybookHistoryRow);
+        }
+    },
+    renderRefreshIcon() {
+        let { refreshing } = this.state;
+        let controlsClass = "glyphicon glyphicon-refresh";
+        let style = {float: 'right', color: 'lightgray'};
+        if (refreshing) {
+            controlsClass += " refreshing"
+            style.color = "inherit";
+        }
+
+        return (
+        <span className={controlsClass} style={style} onClick={this.onRefresh} />
+        );
+    },
+    //For refresh icon
+    componentDidMount() {
+        let {InstancePlaybookStore} = this.props.subscriptions;
+        InstancePlaybookStore.addChangeListener(this.requestListener);
+    },
+    componentWillUnmount() {
+        let {InstancePlaybookStore} = this.props.subscriptions;
+        InstancePlaybookStore.addChangeListener(this.requestListener);
+    },
+
+    requestListener() {
+        this.setState({
+            refreshing: false
+        });
+    },
+
+    onRefresh() {
+        this.setState({
+            refreshing: true
+        });
+        let {InstancePlaybookStore} = this.props.subscriptions;
+
+        InstancePlaybookStore.clearCache();
+        InstancePlaybookStore.getAll();
+    },
+    //End-For refresh icon
+
+    render: function() {
+        if(!featureFlags.hasInstanceSharing()) {
+            return null;
+        }
+        if(! context.profile.get('is_staff') && !context.hasEmulatedSession()) {
+            return null;
+        }
+        return (
+        <div>
+            <div className="resource-details-section section">
+                <h4 className="t-title">Instance Command History</h4>
+                <div style={{maxWidth: "600px"}}>
+                    <p className="alert alert-warning">
+                        <Glyphicon name="info-sign" />
+                    NOTE: This table is NOT visible to regular users.
+                    Staff users will see this for their accounts as well
+                    as accounts that they have emulated.
+                    </p>
+                    <p>
+                        Use the table below to track the commands that
+                        have been executed in your instance.
+                    </p>
+                </div>
+            </div>
+            <div id="container" className="instance-playbook-history">
+                {this.renderRefreshIcon()}
+                <table className="clearfix table" style={{ tableLayout: "fixed" }}>
+                    <thead>
+                        {this.renderHeaderRow()}
```

Comment:
We will likely want to change the look here, too. Example render in screenshot:
<img width="886" alt="screen shot 2017-11-21 at 3 40 54 pm" src="https://user-images.githubusercontent.com/2889930/33100680-7e9d3f9c-ced2-11e7-855e-c7ba480a4acf.png">

Reply:
The tags list has an example on how to format the status to not break to new line. I should make a component for the status.